src/notebooks/R Mia Examples/Comparative-Metagenomics.qmd

```diff
@@ -0,0 +1,852 @@
+```
+
+2. Remove singletons. Singletons are OTUs of size one, meaning that only one read was assigned to that OTU. These very low-abundance OTUs could be biologically real (belonging to the rare biosphere ([1](#ref-lynch2015))), or they could be false positives due to sequencing artefacts. Singletons observed in only one sample are more likely to be artefacts, and it is good practice to remove them from the OTUs counts table to avoid artificially over-estimating the OTUs richness. You can find more discussion about this in [Robert Edgar’s blog](https://drive5.com/usearch/manual/singletons.html).
+
+```{r}
+# Calculate the total counts for each taxa across all samples in the filtered data
+taxa_sums_filtered <- rowSums(assay(tse)[, rownames(filtered_data)])
+
+# Filter taxa to retain only those with a sum greater than 1
+filtered_taxa_indices <- which(taxa_sums_filtered > 1)
+
+# Subset the tse object to keep only non-singleton taxa in the filtered data
+filtered_tse <- tse[filtered_taxa_indices, rownames(filtered_data)]
+```
+
+3. Show some stats on sequencing depth across samples
+
+```{r}
+# Calculate max difference
+sample_sums <- colSums(assay(filtered_tse))
+max_difference <- max(sample_sums) / min(sample_sums)
+sprintf("The max difference in sequencing depth is %s", max_difference)
+
+# Define boxplot data
+boxplot_data <- data.frame(SampleSums = sample_sums)
+
+# Set options
+theme_custom <- theme_minimal() +
+  theme(
+    plot.title = element_text(hjust = 0.5, size = 14, face = "bold"),  # Center the title, increase size, and make bold
+    axis.title.y = element_text(size = 14),  # Make y-axis label same size and font as title
+    axis.text.y = element_text(size = 12),  # Adjust y-axis text size
+    axis.text.x = element_blank(),  # Remove x-axis text
+    axis.ticks.x = element_blank()  # Remove x-axis ticks
+  )
+
+# Plot boxplot
+boxplot_plot <- ggplot(boxplot_data, aes(x = "", y = SampleSums)) +
+  geom_boxplot(fill = "#FF9999") +   # Use a lighter color
+  labs(title = "Sequencing depth across samples", x = "", y = "Number of reads") +   # Add titles and labels
+  theme_custom  # Apply custom theme
+
+# Extract boxplot statistics
+boxplot_stats <- boxplot.stats(sample_sums)$stats
+
+# Create a data frame for the statistics
+stats_df <- data.frame(
+  x = 1,  # Single x-position for the boxplot
+  y = boxplot_stats,  # The y-values are the boxplot stats
+  label = boxplot_stats  # The labels to be printed
+)
+
+# Format the numbers for better appearance
+stats_df$label <- format(stats_df$label, big.mark = ",", scientific = FALSE)
+
+# Adjust vertical and horizontal justification for each label
+stats_df$vjust <- c(-0.5, -0.5, -0.5, 1.5, 1.5)
+stats_df$hjust <- c(1.2, 0.5, 0.5, 0.5, 1.2)
+
+
+# Add text labels for the statistics with adjusted vjust
+boxplot_plot <- boxplot_plot +
+  geom_text(data = stats_df, aes(x = x, y = y, label = label, vjust = vjust, hjust = hjust), size = 3.5)
```

Comment:
I would to boxplot and stats separately, and using the miaViz / scater functionality.

Reply:
Examples can be simplified also, not everything has to match 1-to-1 to the old phyloseq examples.